### PR TITLE
[FLINK-13157]reeanble unit test read complext type of HiveInputFormatTest

### PR DESCRIPTION
**Pull request by zjuwangg**
## What is the purpose of the change

*reeanble unit test read complext type of HiveInputFormatTest*

## Brief change log
  - *reeanble unit test read complext type of HiveInputFormatTest*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no )
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by zjuwangg** [comment]
cc @bowenli86 @xuefuz @lirui-apache 
Travis-Link[Link](https://travis-ci.com/zjuwangg/flink/builds/118422789)

**Comment by bowenli86** [comment]
@zjuwangg thanks for your contribution!

I wonder if we should add a shim for the changes you made in `HiveInspectors`?

**Comment by zjuwangg** [comment]
> @zjuwangg thanks for your contribution!
> 
> I wonder if we should add a shim for the changes you made in `HiveInspectors`?

@bowenli86 I talked with @lirui-apache offline and found a way to make the change in `HiveInspectors` look more reasonable. 
We don't need a shim for the changes right now, maybe it's needed for the long term.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveInputFormatTest.java [hunk 1]
```diff
@@ -125,54 +128,54 @@ public void testReadFromHiveInputFormat() throws Exception {
...
+		StorageDescriptor sd = new StorageDescriptor();
+		String location = HiveInputFormatTest.class.getResource("/complex_test").getPath();
+		sd.setLocation(location);
+		sd.setInputFormat(DEFAULT_HIVE_INPUT_FORMAT_TEST_INPUT_FORMAT_CLASS);
+		sd.setOutputFormat(DEFAULT_OUTPUT_FORMAT_CLASS);
+		sd.setSerdeInfo(new SerDeInfo());
+		sd.getSerdeInfo().setSerializationLib(DEFAULT_HIVE_INPUT_FORMAT_TEST_SERDE_CLASS);
+		sd.getSerdeInfo().setParameters(new HashMap<>());
+		sd.getSerdeInfo().getParameters().put("serialization.format", "1");
+		sd.getSerdeInfo().getParameters().put("field.delim", ";");
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveInputFormatTest.java]: Can we use these config keys from `serdeConstants`?

**Comment by lirui-apache** [comment]
LGTM overall. Probably a follow up is to use hive runner to simplify these tests.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveInputFormatTest.java (hunk 1 above)
Comment (zjuwangg) [reply flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/batch/connectors/hive/HiveInputFormatTest.java]: Good point!

**Comment by zjuwangg** [comment]
address @lirui-apache 's comments.

**Comment by bowenli86** [comment]
> The latest changes look good to me. It would be good if we can revert unnecessary changes on the indention. (This might be taken care at commit time.(

Yes, as @xuefuz mentioned, please revert any unnecessary changes in the future. I reverted them for you this time.

**Comment by zjuwangg** [comment]
> > The latest changes look good to me. It would be good if we can revert unnecessary changes to the indention. (This might be taken care at commit time. (
> 
> Yes, as @xuefuz mentioned, please revert any unnecessary changes in the future. I reverted them for you this time.
> 
> Ran tests for Hive 2.3.4 and 1.2.1 locally. Merging

Thanks, I will take care of in the future.

**Pull request closed without merging**